src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/StringLiteralEqualityCheck.java

```diff
@@ -46,7 +51,18 @@
  * <pre>
  * &lt;module name=&quot;StringLiteralEquality&quot;/&gt;
  * </pre>
+ * <p>
+ * Examples of violations:
+ * </p>
+ * <pre>
+ * if (status == "done") {} // violation
  *
+ * boolean flag = (status == "done") // violation
+ *
+ * boolean flag = (status.equals("done")) // OK
+ *
+ * if (name == getName("x")) {} // OK
```

Comment:
It is not clear what "OK" means here as user sees no string except in the method parameter. I would remove the parameter to not confuse the user, keep the method call and name, and add to "OK"
`limitation that check cannot tell runtime type returned from method call`.

Reply:
It may help to show with context status and name are string variables.